api_tests/lib/actor.ts

```diff
@@ -1,39 +1,72 @@
-const Toml = require("toml");
-const wallet = require("./wallet.js");
-const fs = require("fs");
-const bitcoin = require("./bitcoin.js");
-const ethutil = require("ethereumjs-util");
-
-// TODO: pass value instead of using global config
-const bitcoin_rpc_client = bitcoin.createClient(
-    global.harness.ledgers_config.bitcoin
-);
-
-class Actor {
-    constructor(name, config) {
-        const node_config = global.harness.config.comit_node[name];
-        if (!node_config) {
-            throw new Error("comit_node." + name + " configuration is needed");
-        }
+import { WalletConfig, Wallet } from "./wallet";
+import * as chai from "chai";
+import {
+    Action,
+    SwapResponse,
+    ComitNodeConfig,
+    MetaComitNodeConfig,
+} from "./comit";
+import * as bitcoin from "./bitcoin";
+import * as toml from "toml";
+import * as fs from "fs";
+
+import chaiHttp = require("chai-http");
+
+chai.use(chaiHttp);
+
+export interface BtsieveForComitNodeConfig {
+    poll_interval_secs: number;
+}
+
+export interface TestConfig {
+    comit_node: { [key: string]: MetaComitNodeConfig };
+}
+
+export class Actor {
+    name: string;
+    host: string;
+    wallet: Wallet;
+    comitNodeConfig: ComitNodeConfig;
+
+    constructor(
+        name: string,
+        testConfig?: TestConfig,
+        root?: string,
+        walletConfig?: WalletConfig
+    ) {
         this.name = name;
-        this.host = node_config.host;
-        this.config = Toml.parse(
-            fs.readFileSync(
-                global.harness.test_root +
-                    "/" +
-                    node_config.config_dir +
-                    "/default.toml",
-                "utf8"
-            )
-        );
-        this.wallet = wallet.create(name, config);
+        if (testConfig) {
+            const metaComitNodeConfig = testConfig.comit_node[name];
+            if (!metaComitNodeConfig) {
+                throw new Error("comit_node configuration is needed");
+            }
+
+            this.host = metaComitNodeConfig.host;
+            this.comitNodeConfig = toml.parse(
+                fs.readFileSync(
+                    root +
+                        "/" +
+                        metaComitNodeConfig.config_dir +
+                        "/default.toml",
+                    "utf8"
+                )
+            );
+        }
+
+        if (walletConfig) {
+            this.wallet = new Wallet(name, walletConfig);
+        }
     }
 
     comit_node_url() {
-        return "http://" + this.host + ":" + this.config.http_api.port;
+        return "http://" + this.host + ":" + this.comitNodeConfig.http_api.port;
     }
 
-    poll_comit_node_until(chai, location, predicate) {
+    // `chai` used to be passed down. Not sure why?
```

Comment:
:x: This comment seems to be pretty useless :)

Regarding the actual comment: It is kind of dirty that this lib depends on `chai` since that means it knows it will be executed from within a test.

Reply:
I think @D4nte got rid of this in his consolidation branch anyway.

Reply:
The comment should be removed and now is.

I do not think there is much choice here. The request needs to be done with `chai` in any case.